qualtran/cirq_interop/_cirq_interop.py

```diff
@@ -67,11 +73,24 @@ def short_name(self) -> str:
 
     @cached_property
     def signature(self) -> 'Signature':
-        return Signature([Register('qubits', 1, shape=(self.n_qubits,))])
+        return signature_from_cirq_registers(self.cirq_registers)
 
     @cached_property
-    def n_qubits(self):
-        return cirq.num_qubits(self.gate)
+    def cirq_registers(self) -> cirq_ft.Registers:
+        if isinstance(self.gate, cirq_ft.GateWithRegisters):
+            return self.gate.registers
+        else:
+            return cirq_ft.Registers.build(qubits=cirq.num_qubits(self.gate))
+
+    def decompose_bloq(self) -> 'CompositeBloq':
+        quregs = self.signature.get_cirq_quregs()
```

Comment:
what do you think about moving `get_cirq_quregs` out of `Signature` and into this file?

Reply:
Let's do that in a follow-up PR since `get_cirq_quregs` is also used in a few other places like tests basic gates / composite gates etc.